bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,229 @@
+spec:
+  azureName: #@ dbName
+  owner:
+    name: #@ flexibleServerName
+  charset: #@ db.charset
+  collation: #@ db.collation
+#@ end
+
+#@ fwRuleCounter = 0
+#@ for val in data.values.firewall_rules:
+#@   startAddr = ip.parse_addr(val.startIp)
+#@   endAddr = ip.parse_addr(val.endIp)
+--- 
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersFirewallRule
+metadata:
+  name: #@ data.values.resource_name + "-" + str(fwRuleCounter)
+  namespace: #@ data.values.resource_namespace
+spec:
+  owner:
+    name: #@ flexibleServerName
+  startIpAddress: #@ startAddr.string()
+  endIpAddress: #@ endAddr.string()
+#@   fwRuleCounter += 1
+#@ end  
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServer
+metadata:
+  name: #@ flexibleServerName
+  namespace: #@ data.values.resource_namespace
+spec:
+  location: #@ data.values.location
+  azureName: #@ flexibleServerName
+  owner:
+    name: #@ resourceGroupK8SName
+  version: #@ data.values.version
+  sku:
+    name: #@ data.values.instance_type
+    tier: #@ data.values.instance_tier
+  administratorLogin: #@ data.values.administrator_name
+  administratorLoginPassword:
+    name: #@ dbSecretName
+    key: password
+  storage:
+    storageSizeGB: #@ data.values.instance_storage_size_gb
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - dbforpostgresql.azure.com
+  resources:
+  - flexibleservers
+  - flexibleserversdatabases
+  - flexibleserversfirewallrule
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - resources.azure.com
+  resources:
+  - resourcegroups
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: #@ roleName
+subjects:
+- kind: ServiceAccount
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: secretgen.carvel.dev/v1alpha1
+kind: SecretTemplate
+metadata:
+  name: #@ data.values.resource_name + "-bindable"
+  namespace: #@ data.values.resource_namespace
+spec:
+  serviceAccountName: #@ serviceAccountName
+  inputResources:
+  - name: server
+    ref:
+      apiVersion: dbforpostgresql.azure.com/v1alpha1api20210601
+      kind: FlexibleServer
+      name: #@ flexibleServerName
+  - name: db
+    ref:
+      apiVersion: dbforpostgresql.azure.com/v1alpha1api20210601
+      kind: FlexibleServersDatabase
```

Comment:
*question* This creates a secret for a single database but I see that we allow the user to specify multiple Secrets - therefore we would need to generate multiple Secrets.

For the Google CloudSQL we had a similar conversation around composition of these 1-n resources, we decided for the purpose of building a simple demo-able package we would stick with 1 server and 1 database and therefore 1 Secret.

Reply:
We removed the multiple database feature, as discussed elsewhere. So closing this.